tensorrt_llm/_torch/pyexecutor/resource_manager.py

```
@@ -299,57 +298,34 @@ def append_to_kv_heads_per_layer(num_kv_heads_per_layer: List[int],
-            # to all window sizes as they are full attentions is enough.
-            self.blocks_in_primary_pool = int(kv_cache_config.max_tokens //
-                                              tokens_per_block)
-
-            host_cache_size = kv_cache_config.host_cache_size if kv_cache_config.host_cache_size else 0
-            max_tokens_secondary = host_cache_size // self.get_cache_bytes_per_token(
+        if self.is_vswa:
+            # VSWA case: use C++ implementation for variable window sizes
+            if model_config is None:
+                raise ValueError(
+                    "model_config is required for VSWA (Variable Sliding Window Attention)"
+                )
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)